scripts/iarc_task_action_server.py

```diff
@@ -0,0 +1,143 @@
+#!/usr/bin/env python
+import threading
+import rospy
+import actionlib
+from takeoff_task import TakeoffTask
+from test_task import TestTask
+from iarc7_motion.msg import QuadMoveAction, QuadMoveResult
+
+class IarcTaskActionServer:
+    def __init__(self):
+        self._action_name = "motion_planner_server"
+
+
+        self.action_server = actionlib.ActionServer(self._action_name,
+                                          QuadMoveAction,
+                                          self.new_goal,
+                                          cancel_cb=self.cancel_request,
+                                          auto_start = False)
+        self.goal_tasks = []
+        self.current_task = None
+        self.current_goal = None
+        self.cancel_requested = False
+        self.lock = threading.RLock()
+        # Start action server last to avoid race condition
+        self.action_server.start()
+
+    # Private method
+    def new_goal(self, goal):
+        with self.lock:
+            rospy.logdebug("new_goal: %s", goal.get_goal_id().id)
+
+            task_request = goal.get_goal()
+
+            if task_request.movement_type == "takeoff" :
```

Comment:
This creates a side affect because initially I was thinking that what each task was passed on instantiation could be adjusted. Fixing this should be a new trello card, it requires restructuring of the action and the tasks.